fix: Android SDK 버전 수정 및 iOS 임포트 오류 수정

chlee1001 opened the pull request:
그동안 patch를 이용하여 사용했지만, 직접 포크 된 레포를 가져와 사용하기위해 PR 남깁니다.

chlee1001 committed:
🔧 chore(build.gradle): Gradle, SDK 버전 업데이트
🔧 chore(build.gradle): 네이버 맵 SDK 버전 업데이트
Gradle 버전을 3.4.1에서 3.4.3으로 업데이트하였습니다. 또한, compileSdkVersion, buildToolsVersion, targetSdkVersion을 각각 28에서 33으로 업데이트하였습니다. 이러한 변경은 최신 안드로이드 버전에 대한 호환성을 보장하기 위함입니다. 또한, 네이버 맵 SDK 버전을 3.12.0에서 3.17.0으로 업데이트하여 최신 기능을 사용할 수 있게 하였습니다.

chlee1001 committed:
✨ 피트(index.tsx): PropsWithChildren 타입을 NaverMapView 컴포넌트에 적용
NaverMapView 컴포넌트의 children prop이 Element 타입에서 PropsWithChildren 타입으로 변경되었습니다. 이 변경으로 인해 NaverMapView 컴포넌트는 이제 자식 요소를 더 유연하게 처리할 수 있습니다. 이는 컴포넌트의 재사용성을 향상시키고, 코드의 가독성을 높이며, 더 많은 컴포넌트 구성을 가능하게 합니다.

chlee1001 committed:
🔀 refactor(reactNativeNMap): NMapsMap에서 NMapsGeometry로 NMGLatLng 임포트 경로 변경
NMapsMap에서 제공하는 NMGLatLng 대신 NMapsGeometry에서 제공하는 NMGLatLng를 사용하도록 변경하였습니다. 이 변경은 더 정확한 위치 계산을 가능하게 하며, 더욱 효율적인 코드 구조를 제공합니다. 이로 인해 앱의 전반적인 성능 향상이 기대됩니다.

🔧 refactor(reactNativeNMap): NMGLatLng 임포트 경로 변경
NMapsMap에서 NMapsGeometry로 NMGLatLng의 임포트 경로를 변경했습니다. 이 변경은 라이브러리 업데이트로 인한 경로 변경에 대응하기 위함입니다. 이로 인해 앱은 최신 라이브러리 버전과 호환성을 유지하면서도 정상적으로 작동합니다.

chlee1001 committed:
🔧 refactor(reactNativeNMap): 이벤트 핸들러 이름 변경
이벤트 핸들러의 이름을 좀 더 명확하게 변경하였습니다. 이전에는 onClick이라는 일반적인 이름을 사용했지만, 이제는 onCircleClick, onMarkerClick, onPathClick, onPolygonClick, onPolylineClick 등으로 변경하여 각 이벤트 핸들러가 어떤 동작에 반응하는지 명확하게 알 수 있습니다. 이 변경으로 인해 코드의 가독성이 향상되었습니다.

chlee1001 committed:
🔧 refactor(RNNaverMapView.java, RNNaverMapViewContainer.java, RNNaverMapViewManager.java): 안드로이드 - 현재 위치가 표시되지않는 버그 수정, 위치 권한 요청 코드 및 NaverMapSdk 인스턴스 추가
위치 권한 요청 코드를 상수로 추가하고, NaverMapSdk 인스턴스를 생성하여 맵 캐시를 정리하는 기능을 추가했습니다. 이는 맵의 성능을 향상시키고 사용자에게 더 나은 경험을 제공하기 위함입니다. 또한, FusedLocationSource의 인스턴스 생성을 RNNaverMapView의 생성자로 이동하여 코드의 일관성을 높였습니다. 이러한 변경으로 인해 RNNaverMapViewContainer와 RNNaverMapViewManager의 생성자에서 locationSource 인자를 제거했습니다.

chlee1001 committed:
🔥 refactor(RNNaverMapView.java, RNNaverMapViewContainer.java, RNNaverMapViewManager.java): 불필요한 import 제거 및 Log import 추가
불필요한 import를 제거하여 코드의 가독성을 향상시켰습니다. 또한, Log를 import하여 디버깅을 용이하게 하였습니다. 이러한 변경은 코드의 효율성을 높이고 유지 관리를 용이하게 합니다.

chlee1001 committed:
🐛 fix(RNNaverMapCircleOverlay.h): onCircleClick 속성 선언에 세미콜론 추가
onCircleClick 속성 선언에 세미콜론이 누락되어 컴파일 오류가 발생했습니다. 이를 수정하여 프로젝트의 안정성을 향상시켰습니다.

chlee1001 committed:
🔖 버전(package.json): 패키지 버전을 0.0.66에서 0.0.67로 업데이트
👥 저자(package.json): 저자를 QuadFlask에서 chlee1001로 변경
🔀 홈페이지 및 저장소 URL(package.json): 홈페이지 및 저장소 URL을 chlee1001의 깃허브로 변경

패키지 버전을 업데이트하여 새로운 기능이나 수정 사항을 반영하였습니다. 저자 정보와 홈페이지, 저장소 URL을 변경하여 프로젝트의 소유권을 새로운 저자인 chlee1001에게 이전하였습니다. 이는 프로젝트의 유지 보수와 관리를 위한 변경 사항입니다.

chlee1001 committed:
🔥 remove(RNNaverMapViewManager.java): FusedLocationSource 인스턴스 생성 제거
FusedLocationSource 인스턴스 생성이 제거되었습니다. 이 변경은 불필요한 위치 소스 인스턴스화를 방지하고, 앱의 메모리 사용량을 줄이며, 앱의 성능을 향상시키기 위해 이루어졌습니다.

chlee1001 committed:
🐛 fix(README.md, index.tsx): 'contentsRegion'을 'contentRegion'으로 수정
✨ feat(index.tsx): MarkerProps에 children 속성 추가
'contentsRegion'이라는 오타를 수정하여 'contentRegion'으로 변경하였습니다. 이는 코드의 일관성을 유지하고 혼동을 방지하기 위함입니다. 또한, MarkerProps에 children 속성을 추가하여 Marker 컴포넌트가 자식 요소를 가질 수 있도록 하였습니다. 이는 Marker 컴포넌트의 확장성을 향상시키기 위한 변경입니다.

chlee1001 committed:
🐛 fix(RNNaverMapView.java): IndexOutOfBoundsException 방어 코드 추가
기존에는 features 리스트에서 인덱스를 가져올 때 인덱스 범위를 확인하지 않아 IndexOutOfBoundsException이 발생할 수 있었습니다. 이를 방지하기 위해 인덱스 범위를 확인하는 방어 코드를 추가하였습니다. 이로써, 인덱스가 범위를 벗어날 경우 null을 반환하도록 하여 예외 발생을 방지합니다.

chlee1001 committed:
🔀 refactor(react-native-nmap.podspec): 저자 및 홈페이지 정보 업데이트
이 변경은 패키지 메타데이터를 업데이트하여 새로운 저자의 이메일과 깃허브 주소를 반영합니다. 이는 패키지 소유권이 변경되었음을 나타내며, 사용자가 새로운 저자에게 문의하거나 패키지에 대한 최신 정보를 얻을 수 있도록 합니다.

ZeroCho commented:
최고입니다! 다만 머지가 언제될 지는 모르겠네요 ㅠㅠ

chlee1001 commented:
> 최고입니다! 다만 머지가 언제될 지는 모르겠네요 ㅠㅠ

감사합니다! 유지보수가 안되어서 아쉽습니다 ㅠ

ZeroCho commented:
@chlee1001 한 가지 추가하자면
package.json에서 다음 부분 >16이랑 >60 등으로 업데이트해서
최신 버전 이용자들도 설치 시 오류 안 나게 하면 좋을 것 같습니다~

"peerDependencies": {
    "react": "16.8.6",
    "react-native": "0.60.5"
  },

chlee1001 commented:
> @chlee1001 한 가지 추가하자면 package.json에서 다음 부분 >16이랑 >60 등으로 업데이트해서 최신 버전 이용자들도 설치 시 오류 안 나게 하면 좋을 것 같습니다~
> 
> "peerDependencies": { "react": "16.8.6", "react-native": "0.60.5" },

@ZeroCho 
조언 감사합니다! 
#128 참고하여 업데이트하겠습니다! 👍

chlee1001 committed:
🔧 chore(package.json): 패키지 버전 업데이트 및 종속성 업데이트 패키지 버전을 0.0.67에서 0.0.68로 업데이트하였습니다. 또한, React와 React Native의 peerDependencies를 각각 ">=17", ">=0.66"으로 업데이트하여 최신 버전의 React Native를 지원하도록 했습니다. devDependencies도 최신 버전으로 업데이트하였씁니다.

chlee1001 committed:
🔧 refactor(RNNaverMapView.java): setCenter 메소드에 zoom 파라미터 추가
🔧 refactor(RNNaverMapViewContainer.java): setCenter 메소드에 zoom 파라미터 추가
🔧 refactor(RNNaverMapViewManager.java): animateToCoordinate 메소드에 zoom 파라미터 추가
🔧 refactor(RNNaverMapViewProps.java): setCenter 메소드에 zoom 파라미터 추가
🔧 refactor(index.d.ts): animateToCoordinate 메소드에 zoom 파라미터 추가
setCenter 및 animateToCoordinate 메소드에 zoom 파라미터를 추가하여 지도의 중심을 설정하거나 애니메이션으로 이동할 때 줌 레벨을 동시에 설정할 수 있게 되었습니다. 이 변경으로 인해 사용자는 지도의 위치와 줌 레벨을 한 번에 조정할 수 있게 되어 사용성이 향상되었습니다.
zoom 파라미터를 입력하지 않을 경우, 각 OS SDK 의 기본 값을 따라갑니다.

🔧 fix(index.js, index.tsx, RNNaverMapViewManager.m): animateToCoordinate 함수에 줌 레벨 인자 추가
🔖 chore(package.json): 패키지 버전 0.0.68에서 0.0.69로 업데이트

animateToCoordinate 함수에 줌 레벨 인자를 추가하여, 좌표 이동 시 줌 레벨을 동시에 조정할 수 있게 되었습니다. 이 변경으로 인해 사용자는 지도의 특정 좌표로 이동하면서 동시에 원하는 줌 레벨로 지도를 보는 경험을 할 수 있게 되었습니다. 또한, 패키지 버전을 업데이트하여 이 변경사항을 반영하였습니다.

chlee1001 committed:
:memo: docs(README.md): 업데이트 된 animateToCoordinate 메서드의 파라미터를 반영했습니다.

chlee1001 committed:
🐛 fix(RNNaverMapViewManager.java): 누락된 닫는 중괄호 추가

chlee1001 committed:
🔧 refactor(RNNaverMapCircleOverlay.java): onClick 메소드 오버라이드 추가

RNNaverMapCircleOverlay 클래스에 onClick 메소드를 오버라이드하여 사용자가 지도의 원 오버레이를 클릭했을 때 언제나 NaverMapView 의 onMapClick 핸드러를 호출하도록 수정했습니다.

RyuWoong commented:
덕분에 편안하게 사용할 수 있게 되었습니다. 감사합니다. 🫡

chlee1001 committed:
:wrench: chore: package-lock.json 파일 업데이트

chlee1001 committed:
📝 (README.md): 네이버 지도 저장소 URL 업데이트
🔧 (build.gradle): 네이버 지도 저장소 URL 및 SDK 버전 업데이트

네이버 지도 SDK의 저장소 URL이 jfrog에서 repository로 변경되었습니다. 이를 반영하여 README.md와 build.gradle 파일의 URL을 업데이트했습니다. 또한, 네이버 지도 SDK 버전을 3.17.0에서 3.19.1로 업그레이드하여 최신 기능과 버그 수정을 반영했습니다.

chlee1001 committed:
⬆️ (package.json): 버전 0.0.69에서 0.0.70으로 업그레이드
버전을 0.0.69에서 0.0.70으로 업그레이드하여 새로운 변경 사항이나 수정 사항을 반영합니다. 이는 배포 및 버전 관리를 용이하게 합니다.
